tests/functions/folding/test_fold_as_wei_value.py

```diff
@@ -0,0 +1,60 @@
+from hypothesis import (
+    given,
+    settings,
+    strategies as st,
+)
+import pytest
+
+from vyper import (
+    ast as vy_ast,
+    functions as vy_fn,
+)
+
+denoms = [x for k in vy_fn.AsWeiValue.wei_denoms.keys() for x in k]
+
+
+st_decimals = st.decimals(
+    min_value=0,
+    max_value=2 ** 32,
+    allow_nan=False,
+    allow_infinity=False,
+    places=10,
+)
+
+
+@pytest.mark.fuzzing
+@settings(max_examples=10, deadline=1000)
+@given(value=st_decimals)
+@pytest.mark.parametrize('denom', denoms)
+def test_decimal(get_contract, value, denom):
+    source = f"""
+@public
+def foo(a: decimal) -> uint256:
+    return as_wei_value(a, '{denom}')
+    """
+    contract = get_contract(source)
+
+    vyper_ast = vy_ast.parse_to_ast(f"as_wei_value({value:.10f}, '{denom}')")
```

Comment:
`18f`, because 18 decimal places is the largest valid length for Ether denominations

Reply:
I chose 10 because we only allow a decimal precision of 10.  If we fold a decimal with a precision of 18 then we're allowing an invalid type to get past.

Reply:
Ah gotcha

Reply:
Note: Change this when https://github.com/vyperlang/vyper/issues/1807 is implemented